lib/events.js

```diff
@@ -172,28 +115,17 @@ EventEmitter.prototype.emit = function emit(type) {
     needDomainExit = true;
   }
 
-  var isFn = typeof handler === 'function';
   len = arguments.length;
-  switch (len) {
-    // fast cases
-    case 1:
-      emitNone(handler, isFn, this);
-      break;
-    case 2:
-      emitOne(handler, isFn, this, arguments[1]);
-      break;
-    case 3:
-      emitTwo(handler, isFn, this, arguments[1], arguments[2]);
-      break;
-    case 4:
-      emitThree(handler, isFn, this, arguments[1], arguments[2], arguments[3]);
-      break;
-    // slower
-    default:
-      args = new Array(len - 1);
-      for (i = 1; i < len; i++)
-        args[i - 1] = arguments[i];
-      emitMany(handler, isFn, this, args);
+  args = new Array(len - 1);
+  for (i = 1; i < len; i++)
```

Comment:
Whoops, didn't see it init up top.

However, in terms of speed, it was recently discussed that `i < arr.length` is superior to `i < len`